sdk/dynamicconfig/metric/controller/push/push.go

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package push
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	sdk "go.opentelemetry.io/contrib/sdk/dynamicconfig/metric"
+
+	"go.opentelemetry.io/contrib/sdk/dynamicconfig/metric/controller/notify"
+	nbasic "go.opentelemetry.io/contrib/sdk/dynamicconfig/metric/controller/notify/basic"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/metric/registry"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	controllerTime "go.opentelemetry.io/otel/sdk/metric/controller/time"
+	"go.opentelemetry.io/otel/sdk/metric/processor/basic"
+)
+
+const fallbackPeriod = 10 * time.Minute
+
+// Controller organizes a periodic push of metric data.
+type Controller struct {
+	lock        sync.Mutex
+	accumulator *sdk.Accumulator
+	provider    *registry.Provider
+	processor   *basic.Processor
+	exporter    export.Exporter
+	lastPeriod  time.Duration
+	quit        chan struct{}
+	done        chan struct{}
+	isRunning   bool
+	timeout     time.Duration
+	clock       controllerTime.Clock
+	ticker      controllerTime.Ticker
+	notifier    notify.Notifier
+	mch         notify.MonitorChannel
+	matcher     *PeriodMatcher
+}
+
+// New constructs a Controller, an implementation of metric.Provider,
+// using the provided exporter and options to configure an SDK with
+// periodic collection.
+func New(selector export.AggregatorSelector, exporter export.Exporter, configHost string, opts ...Option) *Controller {
+	c := &Config{}
+	for _, opt := range opts {
+		opt.Apply(c)
+	}
+	if c.Timeout == 0 {
+		c.Timeout = fallbackPeriod
+	}
+
+	processor := basic.New(selector, exporter)
+	impl := sdk.NewAccumulator(
+		processor,
+		sdk.WithResource(c.Resource),
+	)
+
+	notifier := nbasic.NewNotifier(configHost, c.Resource)
+	mch := notify.MonitorChannel{
```

Comment:
Why not use NewMonitorChannel()?

Reply:
Great catch! Got sloppy and forgot to add that in. Adding now